source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -284,29 +291,57 @@ FaultFilter::delayDuration(const Http::RequestHeaderMap& request_headers) {
   return ret;
 }
 
-absl::optional<Http::Code>
-FaultFilter::abortHttpStatus(const Http::RequestHeaderMap& request_headers) {
+AbortHttpAndGrpcStatus FaultFilter::abortStatus(const Http::RequestHeaderMap& request_headers) {
   if (!isAbortEnabled(request_headers)) {
-    return absl::nullopt;
+    return AbortHttpAndGrpcStatus{absl::nullopt, absl::nullopt};
   }
 
+  auto grpc_status = abortGrpcStatus(request_headers);
+
+  // If gRPC status code is set, then HTTP will be set to Http::Code::OK (200).
+  return grpc_status.has_value()
```

Comment:
The way it's implemented now we prioritize grpc faults over http faults. Not sure whether it's a requirement for your implementation but to me http faults should take a precedence over grpc faults since grpc faults are kind of build on top of http requests -> so say it differently, my intuition tell me that grpc faults can happen only if http request succeeds.

Let's say that I send 
`x-envoy-fault-abort-request: 500` and `x-envoy-fault-abort-grpc-request: 14` HTTP headers as part of my request. In this case, we end up with a request that finishes with a 200 HTTP status code and 14 grpc status code. I wouldn't not expect to receive 200 HTTP status code if I send `x-envoy-fault-abort-request: 500` HTTP header.

What would you say to the idea of changing the order here? Check for the HTTP status code abort header first and look into grpc status code abort header only if the http status code abort header is not present.

Going back to my example, if we change the order and send `x-envoy-fault-abort-request: 500` and `x-envoy-fault-abort-grpc-request: 14` HTTP headers. We end up with a request that fails with 500 HTTP status code, it feels OK that we didn't receive 14 grpc status code since the underlying HTTP request failed.

Reply:
I've changed the precedence order favoring http_status over grpc_status headers. And more importantly, updated the documentation on what to expect during such conflicts.